tutorials/external-metastore-for-dataproc/index.md

```diff
@@ -107,17 +110,67 @@ Create a Cloud Storage bucket for storing test data, run the following command o
     -l ${WAREHOUSE_MULTI_REGION} \
     "gs://${PROJECT}-warehouse" 
 
-### Create Cloud SQL instance
 
+### Create private VPC network
+
+Create user defined cloud networking and peer it with Google services
+
+1.  In Cloud Shell, run the following command to create the VPC network:
+
+        gcloud compute networks create ${VPC_NETWORK_NAME} \
+        --subnet-mode=auto \
+        --bgp-routing-mode=global
+
+1.  In Cloud Shell, run the follwing command to reserve the IPs required for the network:
+
+        gcloud compute addresses create \
+        google-managed-services-${VPC_NETWORK_NAME} \
+        --global \
+        --purpose=VPC_PEERING \
+        --prefix-length=16 \
+        --network=projects/${PROJECT}/global/networks/${VPC_NETWORK_NAME}
+
+1.  In Cloud Shell, run the command to setup VPC peering for Google service on the private VPC network:
+
+        gcloud services vpc-peerings connect \
+        --ranges=google-managed-services-${VPC_NETWORK_NAME} \
+        --network=${VPC_NETWORK_NAME} \
+        --project=${PROJECT}
+
+1.  In Cloud Shell, run the following command to enable VM-to-VM communication for Dataproc clusters and CloudSQL.
 
-Create a MySQL on Cloud SQL instance to be used later to host the Hive metastore database.
+        gcloud compute firewall-rules create allow-all-internal \
+        --network projects/${PROJECT}/global/networks/${VPC_NETWORK_NAME} \
+        --allow=all
```

Comment:
Do these ip-ranges cover all regional vpc? otherwise there may be a need to provide firewall rules per region?

Reply:
As written, the firewall rule opens up access to the VPC to the internet since all traffic from 0.0.0.0/0 was allowed.

When tested earlier, this warning alerted me to the rule:
`WARNING: This cluster is configured to use network '..../hive-network' and its associated firewall rules '[allow-all-internal]' which contains the following potential security vulnerability: 'port 8088 is open to the internet, this may allow arbitrary code execution via the YARN REST API. ...`

Reply:
Done
I added
```
--allow-tcp,udp,icmp
--source-ranges=10.128.0.0/9
```